src/app/hooks/useRbfTransactionData.ts

```diff
@@ -0,0 +1,216 @@
+  rbfTxSummary?: {
+    currentFee: number;
+    currentFeeRate: number;
+    minimumRbfFee: number;
+    minimumRbfFeeRate: number;
+  };
+  rbfRecommendedFees?: {
+    medium?: TierFees;
+    high?: TierFees;
+    higher?: TierFees;
+    highest?: TierFees;
+  };
+  mempoolFees?: RecommendedFeeResponse;
+  isLoading?: boolean;
+};
+
+export const isBtcTransaction = (
+  transaction: BtcTransactionData | StxTransactionData,
+): transaction is BtcTransactionData => transaction?.txType === 'bitcoin';
+
+interface LatestNonceResponse {
+  last_mempool_tx_nonce: number;
+  last_executed_tx_nonce: number;
+  possible_next_nonce: number;
+  detected_missing_nonces: Array<number>;
+}
+
+export async function getLatestNonce(
+  stxAddress: string,
+  network: SettingsNetwork,
+): Promise<LatestNonceResponse> {
+  const baseUrl = network?.address;
+  const apiUrl = `${baseUrl}/extended/v1/address/${stxAddress}/nonces`;
+  return axios.get<LatestNonceResponse>(apiUrl).then((response) => response.data);
+}
+
+interface RawTransactionResponse {
+  raw_tx: string;
+}
+
+export async function getRawTransaction(txId: string, network: SettingsNetwork): Promise<string> {
+  const baseUrl = network?.address;
+  const apiUrl = `${baseUrl}/extended/v1/tx/${txId}/raw`;
+
+  return axios.get<RawTransactionResponse>(apiUrl).then((response) => response.data.raw_tx);
+}
+
+const useRbfTransactionData = (transaction?: BtcTransactionData | StxTransactionData): RbfData => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [rbfData, setRbfData] = useState<RbfData>({});
+  const { accountType, network, selectedAccount, stxAvailableBalance, feeMultipliers } =
+    useWalletSelector();
+  const seedVault = useSeedVault();
+  const btcClient = useBtcClient();
+  const selectedNetwork = useNetworkSelector();
+
+  const fetchStxData = useCallback(async () => {
+    if (!transaction || isBtcTransaction(transaction)) {
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+
+      const fee = microStxToStx(transaction.fee);
+      const txRaw: string = await getRawTransaction(transaction.txid, network);
+      const unsignedTx: StacksTransaction = deserializeTransaction(txRaw);
+
+      const [slow, medium, high] = await estimateTransaction(
+        unsignedTx.payload,
+        undefined,
+        selectedNetwork,
+      );
+
+      let minimumFee = fee.multipliedBy(1.25);
+      let highFee = high.fee;
+      let mediumFee = medium.fee;
+
+      if (!Number.isSafeInteger(minimumFee)) {
+        minimumFee = microStxToStx(Math.ceil(stxToMicrostacks(minimumFee).toNumber()));
+      }
+
+      if (feeMultipliers && highFee > BigInt(feeMultipliers?.thresholdHighStacksFee)) {
+        highFee = feeMultipliers.thresholdHighStacksFee;
+      }
+
+      if (feeMultipliers && mediumFee > BigInt(feeMultipliers?.thresholdHighStacksFee)) {
+        mediumFee = feeMultipliers.thresholdHighStacksFee * 0.75;
+      }
```

Comment:
is there an edge case here, where the minimumFee can be higher than the medium/high Fees?

Reply:
yeah, I think that there could be this edge case

Reply:
> yeah, I think that there could be this edge case. should we disable those options then or calculate them to be higher than the current fee?

the unit tests and logic should follow the user story set out here:

https://www.notion.so/xverseapp/Speed-up-Stacks-transaction-Improve-fee-estimation-4455d27f8bdc4ebfbd94a16d7f18fe12?pvs=4#6e0e607014764f2ab9d6f8b76d677dfe

Reply:
for [SPEEDSTX-EXT-2]

Reply:
to clarify: I don't mind when you move logic into core in this case, since you're going to be working on both clients anyway.

but before we can merge, the feature should satisfy the user stories set out in notion.

since we have really clear requirements/specs now, the most efficient way would be for  the PR author to quickly double check the PR matches the notion (in case the requirements changed or no longer make sense, etc.)

thanks!

Reply:
totally makes sense @teebszet, thanks for noticing that! I updated the logic to follower the user story Case 2 and Case 3
![1 Monosnap Monosnap 2024-01-08 18-24-35](https://github.com/secretkeylabs/xverse-web-extension/assets/36603049/c1640bc7-9a7b-4bd7-94e7-5f2af82c6c07)
![2 Monosnap Monosnap 2024-01-08 18-25-06](https://github.com/secretkeylabs/xverse-web-extension/assets/36603049/9cd1d331-6262-4c14-bba4-edf8074b79f1)